### ISSUE TITLE
[ENH] Add `Metric` support to `ptf-v2`

### ISSUE THREAD

**Issue by phoeenniixx** [sktime/pytorch-forecasting#1956]
In #1844, @PranavBhatP mentioned we should move `loss` from `nn.Module` to `Metric` in `ptf-v2`. 
Right now, the `data_module` returns a 3D tensors for `target` (the ground-truth) in the shape `(batch_size, pred_length, num_targets)`
see the lines `collate_fn` here:
https://github.com/sktime/pytorch-forecasting/blob/3821c0bbca6d35efd00c1bcf51b42da1971b24a4/pytorch_forecasting/data/data_module.py#L732-L734

But @PranavBhatP showed that atleast point metrics uses 2D ground truth.
So, we have to make some changes to either the `Metrics` or `data_module`.
In v1, Multi-target means a list of tensors, while in `data_module` uses a new dimension for `num_targets`.

**Comment by phoeenniixx** [comment]
I think we should update the `data_module` to have lists for multi-target, and make sure `Metric` input expectations matches the model output formats.

FYI @fkiraly, @agobbifbk, @PranavBhatP, @xandie985

**Comment by PranavBhatP** [comment]
> I think we should update the data_module to have lists for multi-target, and make sure Metric input expectations matches the model output formats.

this is the approach I would take, since this will involve almost no change on the metrics side.
